### ISSUE TITLE
highlighter tool for text subjects

### ISSUE THREAD

**Issue by simensta** [zooniverse/Panoptes-Front-End#3458]
## Expected behavior
A highlighter tool for selecting metadata in text subjects.
<img width="921" alt="highlighterExample" src="https://cloud.githubusercontent.com/assets/7684729/22743697/9540f546-ede0-11e6-81db-d3537ae9074c.png">

A volunteer should be able to select a highlighter label. Then, select a portion of text. After the text is selected, the background color for that selection should match the color of the label.

The annotation can be represented by an anchorIndex and a focusIndex.

```
annotation: {
     _toolIndex: 0,
    task: "T1",
    value: {
       0: {
           anchorIndex: 41,
           focusIndex: 49,
           labelInformation: {
                color: "#00f00",
                label: "Sender"
        }
    }
}
```

**Comment by eatyourgreens** [comment]
What do `anchorIndex` and `focusIndex` refer to?

**Comment by simensta** [comment]
The `anchorIndex` is the character index for the beginning of the selection.`focusIndex` is the character index for the end of the selection. I chose the labels based on the [Selection API](https://developer.mozilla.org/en-US/docs/Web/API/Selection). However, it might be clearer to use `start` and `end` as labels instead.

**Comment by eatyourgreens** [comment]
`start` and `end` sound more sensible to me. I suspect researchers using this tool will be confused by the terminology used by the Selection API.

**Comment by eatyourgreens** [comment]
>A volunteer should be able to select a highlighter label. Then, select a portion of text. After the text is selected, the background color for that selection should match the color of the label.

I've found you can't reliably detect when a selection is complete, particularly when selecting text by tapping on iOS or Android, or selecting text from the keyboard with a screenreader. So, you might need an additional action (probably pressing a button) to trigger `document.getSelection()`.

**Comment by eatyourgreens** [comment]
Noting that the UI should allow for deleting an annotation too.

**Comment by simensta** [comment]
Closed by #3738.
